DB/NEW_KT_DB/DataAccess/ObjectManager.py

```diff
@@ -12,34 +12,56 @@ def __init__(self, db_file: str):
     # for internal use only:
 
     # Riki7649255 based on rachel-8511
-    def create_management_table(self, table_name, table_structure='object_id INTEGER PRIMARY KEY AUTOINCREMENT,type_object TEXT NOT NULL,metadata TEXT NOT NULL')
+    def create_management_table(self, table_name, table_structure='object_id INTEGER PRIMARY KEY AUTOINCREMENT,type_object TEXT NOT NULL,metadata TEXT NOT NULL'):
+        self.db_manager.create_table(table_name, table_structure)
+    
+    def create_management_table_with_str_id(self, table_name, table_structure='object_id TEXT NOT NULL PRIMARY KEY ,metadata TEXT NOT NULL'):
         self.db_manager.create_table(table_name, table_structure)
 
     
     # Riki7649255 based on saraNoigershel
-    def insert_object_to_management_table(self, table_name, object):
-        self.db_manager.insert_data_into_table(table_name, object)
+    def insert_object_to_management_table(self, table_name, object_metadata):
+        self.db_manager.insert_data_into_table(table_name, ['metadata'], (object_metadata))
+
+    def insert_object_to_management_table_with_str_id(self, table_name, id, metadata):
+        self.db_manager.insert_data_into_table(table_name, ['object_id','metadata'],(id, metadata))
 
 
     # Riki7649255 based on rachel-8511
     def update_object_in_management_table_by_criteria(self, table_name, updates, criteria):
         self.db_manager.update_records_in_table(table_name, updates, criteria)
+    
+    def update_object_in_management_table_by_id(self, table_name, object_id, updates):
+        self.db_manager.update_records_in_table(table_name, updates, f'object_id = {object_id}')
+
 
 
     # rachel-8511, Riki7649255
-    def get_object_from_management_table(self, object_id: int) -> Dict[str, Any]:
+    def get_object_from_management_table(self, table_name, object_id: int, columns = ["*"]) -> Dict[str, Any]:
         '''Retrieve an object from the database.'''
-        result = self.db_manager.select_and_return_records_from_table(self.table_name, ['type_object', 'metadata'], f'object_id = {object_id}')
+        result = self.db_manager.select_and_return_records_from_table(table_name, columns, criteria= f'object_id = {object_id}')
```

Comment:
same as before, what if pk_column name is different

Reply:
you can change the criteria as you want, It is a default value